### DISCUSSION TITLE
How to add behavior to ComboBox SelectionChanged event

### DISCUSSION THREAD

**Discussion by modula2019** [unoplatform/uno#16856] [Q&A] [answered]
Hello all I have been able to get the `ComboBox` from  `Microsoft.UI.Xaml.Controls.ComboBox` package  working. However, I want to be able to use the interactivity behavior for the `SelectionChanged` event as explained here in this blog https://github.com/unoplatform/Uno.XamlBehaviors. The moment I added the packages for the behavior, the application stopped building. I am using net8.0 and MVUX                                                                                                                                `<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
   <PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed"/>
 </ItemGroup>

 <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
   <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed"/>
 </ItemGroup>`

Sample project added

[UnoApp1.zip](https://github.com/unoplatform/uno/files/15422914/UnoApp1.zip)

Best regards

**Comment by agneszitte** [comment, marked as answer]
Hi @modula2019,
Just so it is easier for anyone searching for the same info, I will re-share here the same info I shared with you on [Discord here](https://discord.com/channels/1182775715242967050/1182775715729526908/1243330084031434762). I noticed after my answer there that you opened a discussion here.

Similar concept to the [WCT v7.1.2 details](https://discord.com/channels/1182775715242967050/1182775715729526908/1237939787176869968) that I previously shared with you in terms of how to add the `PackageReference` and `PackageVersion` with the new single project structure. In attachment here [UnoApp1-updated.zip](https://github.com/unoplatform/uno/files/15425122/UnoApp1-updated.zip), you will find your sample with the adjustments. 

In Directory.Package.props:
 
```
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
    <PackageVersion Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
</ItemGroup>
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
    <PackageVersion Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.4.2" />
</ItemGroup>
```
 

In UnoApp1.csproj:
 
```
  <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
    <PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" />
  </ItemGroup>
  <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" />
  </ItemGroup>
```
 
I also updated `<Interactivity:Interacton.Behaviors>` for the correct `<Interactivity:Interaction.Behaviors>` in your code 🙂

**Reply by agneszitte** [reply]
@jeromelaban I think that we would need to update the details for https://github.com/unoplatform/Uno.XamlBehaviors.

**Reply by modula2019** [reply]
It works. Thanks

**Reply by agneszitte** [reply]
Glad to hear that it works properly with the details @modula2019 :relaxed: !
Here is the related issue for the documentation update: https://github.com/unoplatform/uno/issues/16871
